corehq/apps/accounting/static/accounting/ko/accounting.stripe_card_manager.js

```diff
@@ -0,0 +1,153 @@
+var NewStripeCard = function(data){
+    'use strict';
+    var self = this;
+    var mapping = {
+        observe: ['number', 'cvc', 'expMonth','expYear', 'isAutopay', 'token']
+    };
+
+    self.wrap = function(data){
+        ko.mapping.fromJS(data, mapping, self);
+    };
+    self.reset = function(){
+        self.wrap({'number': '', 'cvc': '', 'expMonth': '', 'expYear': '', 'isAutopay': false, 'token': ''});
+    };
+    self.reset();
+
+    self.unwrap = function(){
+        return {token: self.token(), autopay: self.isAutopay()};
+    };
+
+    self.isTestMode = ko.observable(false);
+    self.isProcessing = ko.observable(false);
+    self.agreedToPrivacyPolicy = ko.observable(false);
+    self.errorMsg = ko.observable('');
+
+    var submit = function(){
+        // Sends the new card to HQ
+        return $.ajax({
+            type: "POST",
+            url: data.url,
+            data: self.unwrap()
+        }).success(function(data){
+            $("#card-modal").modal('hide');
+            $("#success-modal").modal('show');
+            cardManager.wrap(data);
+            self.reset();
+        }).fail(function(data){
+            var response = JSON.parse(data.responseText);
```

Comment:
another minor point: shouldn't this have been a JSON response and thus can access the error from the object?